frontend/catalyst/python_interface/inspection/xdsl_conversion.py

```diff
@@ -158,17 +172,33 @@ def _apply_adjoint_and_ctrls(qml_op: Operator, xdsl_op) -> Operator:
 
 
 # pylint: disable=too-many-return-statements
-def resolve_constant_params(ssa: SSAValue) -> float | int:
+def resolve_constant_params(ssa: SSAValue) -> float | int | str:
     """Resolve a constant parameter SSA value to a Python float or int."""
     op = ssa.owner
 
+    if isinstance(op, Block):
+        arg_name = list(compress(op.args, map(lambda arg: arg is ssa, op.args)))[0]
```

Comment:
This should work, right?
```suggestion
        arg_name = next(compress(op.args, map(lambda arg: arg is ssa, op.args)))
```